### PR TITLE
feat: move configuration and node caches to `~/.ziggurat` directory

### LINKED ISSUE #103: If you don't specify a -datadir argument to zcashd and run the tool, it seems to delete the contents of any ~/.zcash directory that exists (potential loss of funds issue)

### PR DESCRIPTION
**Pull request by niklaslong**
Supersedes #104 and closes #103.

### CODE REVIEW

File: src/setup/config.rs [hunk 1]
```diff
@@ -118,10 +135,10 @@ impl NodeMetaData {
                     ));
                 }
                 start_args.insert(n_args - 1, "--config".into());
-                start_args.insert(n_args, config_path.into_os_string());
+                start_args.insert(n_args, config_file_path.into_os_string());
             }
             NodeKind::Zcashd => {
-                start_args.push(format!("-conf={}", config_path.to_str().unwrap()).into());
+                start_args.push(format!("-datadir={}", config_path.to_str().unwrap()).into());
```
Comment (Mirko-von-Leipzig) [inline src/setup/config.rs:141]: What happens if the user adds his own `-datadir` arg? Which one wins?

**Review by Mirko-von-Leipzig — APPROVED** [review]
LGTM

File: src/setup/config.rs (hunk 1 above)
Comment (niklaslong) [reply src/setup/config.rs:141]: Any user defined `-datadir` arg will get ovveridden by Ziggurat's.

Comment (daira) [reply src/setup/config.rs:141]: This should be documented in the readme.

**Pull request merged**